x-pack/legacy/plugins/lens/mappings.json

```diff
@@ -7,14 +7,13 @@
       "visualizationType": {
         "type": "keyword"
       },
-      "activeDatasourceId": {
-        "type": "keyword"
-      },
       "state": {
-        "type": "text"
+        "enabled": false,
+        "type": "object"
```

Comment:
This is an elasticsearch mapping, right? `state.datasourceMetadata` and all of the other object properties seem important to map individually

Reply:
I think this approach is the right one. It's a blob as far as elasticsearch is concerned. We don't need to dive into the details of it now. If we ever do, we can migrate and extract state as needed.